os/kernel/irq/irq_notify.c

```diff
@@ -0,0 +1,78 @@
+/****************************************************************************
+ *
+ * Copyright 2020 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdbool.h>
+#include <arch/irq.h>
+#include <sys/types.h>
+#include <tinyara/arch.h>
+#include <tinyara/irq.h>
+#include <tinyara/sched.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: irq_notify
+ *
+ * Description:
+ *   This function sends an irq event to specific pid task.
+ *   If that task was in blocked queue, it will be unblocked when receiving irq event.
+ *
+ * Inputs:
+ *   pid - The task ID of the task to receive the irq event.
+ *   data - The data which passed to user.
+ *
+ * Return Value:
+ *   OK on success; or ERROR on failure
+ *
+ ****************************************************************************/
+
+int irq_notify(pid_t pid, int data)
```

Comment:
If this function implements FIN, why don't you use 'FIN' when naming this function?